sklego/meta/grouped_predictor.py

```diff
@@ -409,3 +437,99 @@ def decision_function(self, X):
             return self.__predict_groups(X_group, X_value, method="decision_function")
         else:
             return self.__predict_shrinkage_groups(X_group, X_value, method="decision_function")
+
+    @property
+    def _estimator_type(self):
+        """Computes `_estimator_type` dynamically from the wrapped model."""
+        return self.estimator._estimator_type
+
+
+class GroupedRegressor(GroupedPredictor, RegressorMixin):
+    """`GroupedRegressor` is a meta-estimator that fits a separate regressor for each group in the input data.
+
+    Its spec is the same as [`GroupedPredictor`][sklego.meta.grouped_predictor.GroupedPredictor] but it is available
+    only for regression models.
+
+    !!! info "New in version 0.7.5"
+    """
+
+    def fit(self, X, y):
+        """Fit one regressor for each group of training data `X` and `y`.
+
+        Will also learn the groups that exist within the training dataset.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            Training data.
+        y : array-like of shape (n_samples,)
+            Target values.
+
+        Returns
+        -------
+        self : GroupedRegressor
+            The fitted regressor.
+
+        Raises
+        -------
+        ValueError
+            If the supplied estimator is not a regressor.
+        """
+        if not is_regressor(self.estimator):
+            raise ValueError("GroupedRegressor is only available for regression models")
+
+        return super().fit(X, y)
+
+
+class GroupedClassifier(GroupedPredictor, ClassifierMixin):
+    """`GroupedClassifier` is a meta-estimator that fits a separate classifier for each group in the input data.
+
+    Its equivalent to [`GroupedPredictor`][sklego.meta.grouped_predictor.GroupedPredictor] with `shrinkage=None`
+    but it is available only for classification models.
+
+    !!! info "New in version 0.7.5"
```

Comment:
How do you feel about this?

Proof of rendering:
![image](https://github.com/koaning/scikit-lego/assets/42817048/a9115b45-cac7-408c-8531-aa9a63fd4b08)